src/adapters/mc/OsConfigResource.c

```diff
@@ -723,9 +755,9 @@ void MI_CALL OsConfigResource_Invoke_TestTargetResource(
     }
 
     // Determine compliance
-    if ((in->InputResource.value->DesiredObjectValue.exists == MI_TRUE) && (in->InputResource.value->DesiredObjectValue.value != NULL))
+    if ((in->InputResource.value->ExpectedObjectValue.exists == MI_TRUE) && (in->InputResource.value->ExpectedObjectValue.value != NULL))
     {
-        if (0 == strcmp(in->InputResource.value->DesiredObjectValue.value, g_reportedObjectValue))
+        if (0 == strncmp(in->InputResource.value->ExpectedObjectValue.value, g_reportedObjectValue, strlen(in->InputResource.value->ExpectedObjectValue.value)))

```

Comment:
Does this mean it is only compliant if one value matches exactly? Is there no use case where a whole range of values can be considered compliant?

Reply:
Read again. Hint: no, the values do not have to match entirely (they cannot).

Reply:
It brings me back to why we would prepend PASS/FAIL on the reason phrase, rather than returning a separate value which indicates the compliance. Can you elaborate?

Reply:
Reason phrases do not have PASS/FAILED into them. Those can appear only internally in the strings returned by the reported MIM objects for audit.

Reply:
Exactly, that is why I ask why they are baked into this variable in the first place. Seems that we need to take it out either way before attaching to the resource object? Why not a separate output then for PASS/FAIL? It would be cleaner.

Reply:
It would appear clear here but increase complication everywhere else in the stack if all reported MIJM objects for the security baseline go from simple (1 string MIM setting per object) to complex (2 string MIM settings per object, one to hold "PASS" or "FAIL" and the other the reason). And is really not necessary because as explained, failed values are reasons, passed values are a prefix (that could be changed in the future via the policy MOF and module implementation) followed by reason, where the prefix is used to compare and is not reported with the reason.